src/c++/perf_analyzer/test_concurrency_manager.cc

```diff
@@ -355,7 +457,231 @@ TEST_CASE("concurrency_free_ctx_ids")
   // The first sequence should only be called two times, once at the very start,
   // and once during shutdown
   //
-  CHECK(trrm.stats_->sequence_status.seq_ids_to_count.at(1) == 2);
+  CHECK(tcm.stats_->sequence_status.seq_ids_to_count.at(1) == 2);
+}
+
+TEST_CASE("Concurrency - shared memory infer input calls")
+{
+  PerfAnalyzerParameters params{};
+  params.max_concurrency = 4;
+  bool is_sequence_model{false};
+
+  const auto& ParameterizeAsyncAndStreaming{[&]() {
+    SUBCASE("sync non-streaming")
+    {
+      params.async = false;
+      params.streaming = false;
+    }
+    SUBCASE("async non-streaming")
+    {
+      params.async = true;
+      params.streaming = false;
+    }
+    SUBCASE("async streaming")
+    {
+      params.async = true;
+      params.streaming = true;
+    }
+  }};
+
+  const auto& ParameterizeSequence{[&]() {
+    SUBCASE("non-sequence")
+    {
+      is_sequence_model = false;
+      ParameterizeAsyncAndStreaming();
+    }
+    SUBCASE("sequence")
+    {
+      is_sequence_model = true;
+      params.num_of_sequences = 1;
+      ParameterizeAsyncAndStreaming();
+    }
+  }};
+
+  const auto& ParameterizeMemory{[&]() {
+    SUBCASE("No shared memory")
+    {
+      params.shared_memory_type = NO_SHARED_MEMORY;
+      ParameterizeSequence();
+    }
+    SUBCASE("system shared memory")
+    {
+      params.shared_memory_type = SYSTEM_SHARED_MEMORY;
+      ParameterizeSequence();
+    }
+    SUBCASE("cuda shared memory")
+    {
+      params.shared_memory_type = CUDA_SHARED_MEMORY;
+      ParameterizeSequence();
+    }
+  }};
+
+  ParameterizeMemory();
+
+  TestConcurrencyManager tcm(params, is_sequence_model);
+  std::shared_ptr<MockModelParser> mmp{
```

Comment:
What is the output of all of this code? Just the json_str? Or are mmp/mdl used? Curious now that you have it working if this ugliness can be extracted out into a helper function in the base test class.

Reply:
This part could be extracted, yes.

Reply:
extracted some the code to clean things up